### PR TITLE
[MI-1672] Setup Component

### PR DESCRIPTION
**Pull request by joseconsador**
/cc @zendesk/mintegrations

### Description
Add setup dumb component.
Add SET_SETTINGS action to add the app settings to the state.
Pass zafClient instance to AppContainer.

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-1672

### Risks
* [low] Component will not render.
* [medium] App state may not hold correct values for client->settings

### CODE REVIEW

File: src/Components/App/index.jsx [hunk 1]
```diff
@@ -3,23 +3,39 @@ import PropTypes from 'prop-types';
...
-      ) : (
-        <Setup url="https://google.com" />
-      )}
-    </div>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
Comment (samgavinio) [inline src/Components/App/index.jsx]: Is moving these state variables here temporary only?

File: src/Actions/constants.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const ZAF_REGISTERED = 'ZAF_REGISTERED';
-const FETCH_SETTINGS = 'FETCH_SETTINGS';
+const SET_SETTINGS = 'SET_SETTINGS';
```
Comment (jmramos02) [inline src/Actions/constants.js]: `SET_CONFIG` would be more appropriate?

File: tests/Components/setup.test.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
+import renderer from 'react-test-renderer';
 import React from 'react';
 import Setup from '../../src/Components/Setup';
-import renderer from 'react-test-renderer';
 
 it('Setup component renders correctly', () => {
   const tree = renderer.create(
-    <Setup url="https://google.com" />
+    <Setup url="https://google.com" />,
```
Comment (jmramos02) [inline tests/Components/setup.test.js]: We can use faker to generate random test data 

https://www.npmjs.com/package/faker

Comment (kcasas) [inline tests/Components/setup.test.js]: Can't use faker here cause snapshot test will fail if the generated view is different on the snapshot saved.

File: src/Components/App/index.jsx (hunk 1 above)
Comment (joseconsador) [inline src/Components/App/index.jsx]: @samgavinio Yes, on my other PR (https://github.com/zendesk/shopify_app/pull/98/files#diff-f26998a02ecd0af7b19a02903716650b) I moved them to a separate file, that way we can reuse the initial state for testing. Unless there's a better pattern?

File: src/Actions/constants.js (hunk 2 above)
Comment (joseconsador) [inline src/Actions/constants.js]: 💇‍♂️

File: src/Components/App/index.jsx [hunk 4]
```diff
@@ -3,23 +3,39 @@ import PropTypes from 'prop-types';
...
+  }
 }
 
 App.defaultProps = {
   isLoading: true,
+  setupUrl: '',
 };
 
 App.propTypes = {
+  setupUrl: PropTypes.string.isRequired,
```
Comment (kcasas) [inline src/Components/App/index.jsx]: I think we can use regex to check if the url is a valid setup url.

File: src/Components/App/index.jsx [hunk 5]
```diff
@@ -3,23 +3,39 @@ import PropTypes from 'prop-types';
...
+      isCollapsed: false,
+      orderId: null,
+      isTokenValid: false,
+    };
+  }
+
+  render() {
+    return (
+      <div className="App">
+        { this.props.isLoading ? (
```
Comment (kcasas) [inline src/Components/App/index.jsx]: Is there a reason why you should use `this.props` vs just `props`?

**Comment by joseconsador** [comment]
Removed uglifyjs as it breaks the build when compiling es6. It's currently not supported - created a separate JIRA task to uglify the assets (https://zendesk.atlassian.net/browse/MI-1816) .

**Pull request merged**